h2/src/main/org/h2/engine/Database.java

```diff
@@ -1049,7 +1053,7 @@ public void unlockMetaDebug(Session session) {
      * @param session the session
      * @param id the id of the object to remove
      */
-    public synchronized void removeMeta(Session session, int id) {
+    public void removeMeta(Session session, int id) {
         if (id > 0 && !starting) {
```

Comment:
starting is not volatile or atomic, is this safe now we're touching it without a lock?

Reply:
That field only mutates within db opening, which is performed by a single thread, and only after it's final value of false is reached, other threads may gain access to it, but strictly speaking, it needs to be volatile or something. Chance of a race here is minuscule, but I will put volatile.